### PR TITLE
ISPN-16668 Remove unnecessary conversions with queries

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.redhat.com/browse/ISPN-16668

### CODE REVIEW

**Review by fax4ever — CHANGES_REQUESTED** [review]
Thanks Pedro for this contribution.
It is great to simplify the loading process avoiding some conversion.

In order to be sure that everything works I would rebase this change on top of https://github.com/infinispan/infinispan/pull/12864.
In this case we will be sure 100% that the simplification do not affect any improvements.

I don't know what you think of it.

**Comment by pruivo** [comment]
@fax4ever sure. I want to refactor some methods to remove the duplicated code too.

File: query/src/main/java/org/infinispan/query/impl/BaseEntityLoader.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+   }
+
+   @Override
+   public final List<EntityLoaded<E>> loadBlocking(List<?> identifiers, Deadline deadline) {
+      assert identifiers != null;
+      if (identifiers.isEmpty()) {
+         return List.of();
+      }
+
+      long start = queryStatistics.isEnabled() ? System.nanoTime() : 0;
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/impl/BaseEntityLoader.java:30]: Shouldn't we be using the TimeService for these ?

File: query/src/main/java/org/infinispan/query/impl/EntityLoader.java [hunk 2]
```diff
@@ -17,54 +12,20 @@
...
-   }
-
-   @Override
-   public boolean equals(Object o) {
-      if (this == o) return true;
-      if (o == null || getClass() != o.getClass()) return false;
-      EntityLoader<?> that = (EntityLoader<?>) o;
-      return Objects.equals(cache, that.cache) && Objects.equals(queryStatistics, that.queryStatistics);
+   Map<Object, Object> loadEntries(List<?> keys) {
+      return cache.withStorageMediaType().getAll(Set.copyOf(keys));
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/impl/EntityLoader.java:24]: Do we care about the order here as well? The getAll command technically supports the order of the Set passed in as well.

File: query/src/main/java/org/infinispan/query/impl/MetadataEntityLoader.java [hunk 3]
```diff
@@ -18,57 +13,19 @@
...
    }
 
    @Override
-   public boolean equals(Object o) {
-      if (this == o) return true;
-      if (o == null || getClass() != o.getClass()) return false;
-      MetadataEntityLoader<?> that = (MetadataEntityLoader<?>) o;
-      return Objects.equals(cache, that.cache) && Objects.equals(queryStatistics, that.queryStatistics);
+   Map<Object, CacheEntry<Object, Object>> loadEntries(List<?> keys) {
+      return cache.withStorageMediaType().getAllCacheEntries(Set.copyOf(keys));
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/impl/MetadataEntityLoader.java:24]: Same here about the ordering.

File: query/src/main/java/org/infinispan/query/impl/BaseEntityLoader.java (hunk 1 above)
Comment (pruivo) [reply query/src/main/java/org/infinispan/query/impl/BaseEntityLoader.java:30]: Yes. The query module is still using `System.nanoTime()` everywhere and we haven't updated it yet. I can update it here!

**Comment by pruivo** [comment]
@wburns @fax4ever rebased and updated!

**Review by fax4ever — APPROVED** [review]
@pruivo great job. +1 to merge it from me. Also CI looks good and I run some tests locally. I would say let's integrate it as soon as also @wburns agrees. Thanks

**Comment by fax4ever** [comment]
Merged, thanks for this contribution @pruivo

**Pull request merged**